flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/Buckets.java

```diff
@@ -65,6 +65,9 @@
 
 	private final RollingPolicy<IN, BucketID> rollingPolicy;
 
+	@Nullable
+	private final BucketLifeCycleListener<IN, BucketID> bucketLifeCycleListener;
```

Comment:
Should this be a `List<Listener>`?

Reply:
Do we need to supports more than one listeners currently ? I think we might postpone the support if it could support the current use cases. We should be able to keep compatibility with adding a new `withBucketLifeCycleListener(List<BucketLifeCycleListener> listeners>` to the  format builder.